FailedBonusActivationActivity/handler.ts

```diff
@@ -0,0 +1,197 @@
+    _ => void 0
+  );
+};
+
+const updateBonusAsFailed = (
+  bonusActivationModel: BonusActivationModel,
+  bonusActivation: BonusActivationWithFamilyUID
+): TaskEither<Error, RetrievedBonusActivation> => {
+  return fromQueryEither(() => {
+    const bonusToUpdate: NewBonusActivation = {
+      ...bonusActivation,
+      id: bonusActivation.id as BonusCode & NonEmptyString,
+      kind: "INewBonusActivation",
+      status: BonusActivationStatusEnum.FAILED
+    };
+    return bonusActivationModel.createOrUpdate(
+      bonusToUpdate,
+      bonusActivation.id
+    );
+  });
+};
+
+const deleteEligibilityCheck = (
+  eligibilityCheckModel: EligibilityCheckModel,
+  bonusActivation: BonusActivationWithFamilyUID
+): TaskEither<Error, string> => {
+  return fromQueryEither(() =>
+    eligibilityCheckModel.deleteOneById(
+      bonusActivation.id as BonusCode & NonEmptyString
+    )
+  );
+};
+
+/**
+ * Operations to be perfomed in case the bonus request was rejected by ADE
+ *
+ *
+ * @returns either a success or a failure request
+ * @throws when the response is considered a transient failure and thus is not considered a domain message
+ */
+export function FailedBonusActivationHandler(
+  bonusActivationModel: BonusActivationModel,
+  bonusLeaseModel: BonusLeaseModel,
+  eligibilityCheckModel: EligibilityCheckModel
+): IFailedBonusActivationHandler {
+  return async (
+    context: Context,
+    input: unknown
+  ): Promise<FailedBonusActivationResult> => {
+    return taskEither
+      .of<FailedBonusActivationFailure, void>(void 0)
+      .chain(_ =>
+        fromEither(FailedBonusActivationInput.decode(input))
+          .mapLeft(() => InvalidInputFailure.encode({ kind: "INVALID_INPUT" }))
+          .map(({ bonusActivation }) => bonusActivation)
+      )
+      .chain(bonusActivation =>
+        deleteEligibilityCheck(eligibilityCheckModel, bonusActivation)
+          // just ignore this error
+          .foldTaskEither(
+            err => {
+              context.log.warn(
+                `FailedBonusActivationHandler|WARN|Failed deleting dsu: ${err.message}`
+              );
+              return taskEither.of(bonusActivation);
+            },
+            _ => taskEither.of(bonusActivation)
+          )
+      )
+      .chain(bonusActivation =>
+        updateBonusAsFailed(bonusActivationModel, bonusActivation)
+          // just ignore this error
+          .foldTaskEither(
+            err => {
+              context.log.warn(
+                `FailedBonusActivationHandler|WARN|Failed updating bonus: ${err.message}`
+              );
+              return taskEither.of(bonusActivation);
+            },
+            _ => taskEither.of(bonusActivation)
+          )
+      )
+      .chain(bonusActivation =>
+        relaseLockForUserFamily(
+          bonusLeaseModel,
+          bonusActivation.familyUID
+        ).mapLeft(err => {
+          context.log.warn(
```

Comment:
when `relaseLockForUserFamily` the activity must throw in order to trigger a retry